pkg/target/target.go

```diff
@@ -565,6 +580,81 @@ func Summary(targets []*Target) fleet.BundleSummary {
 	return bundleSummary
 }
 
+// tplFuncMap returns a mapping of all of the functions from sprig but removes potentially dangerous operations
+func tplFuncMap() template.FuncMap {
+	f := sprig.TxtFuncMap()
+	delete(f, "env")
+	delete(f, "expandenv")
+	delete(f, "include")
+	delete(f, "tpl")
+
+	return f
+}
+
+func processTemplateValues(valuesMap map[string]interface{}, templateContext map[string]interface{}) (map[string]interface{}, error) {
+	tplFn := template.New("values").Funcs(tplFuncMap()).Option("missingkey=error")
+	tplResult, err := templateSubstitutions(valuesMap, templateContext, tplFn)
+	if err != nil {
+		return nil, err
+	}
+	compiledYaml, ok := tplResult.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("templated result was expected to be map[string]interface{}, got %T", tplResult)
+	}
+
+	return compiledYaml, nil
+}
+
+func templateSubstitutions(src interface{}, templateContext map[string]interface{}, tplFn *template.Template) (interface{}, error) {
```

Comment:
Without knowing too much about sprig, do we have to walk the values and apply the template to every value?
Could we just treat the values as one big template? E.g. marshap It would allow keys to be templated, but otherwise?

Reply:
sprig is simply providing an additional set of functions inside the templating environment, behind the scenes we're just  using `text/template` to do the actual templating. I had an iteration where I did exactly what you suggested where I would simply serialize the entire `values:` key and pass it through the template engine, however I reverted to explicit templating of keys and values because I worried about the security implications. A carefully crafted template could alter the structure of the object once templating is finished. In this approach, we're ensuring the that all templating is scoped to that particular field. One drawback with this approach right now is that it only let's you template input strings to output strings. No way to output `null`, `integer`, or `float` yaml types.

Reply:
Hm, I see. If we let the template instructions from the cluster resource operate on the whole existing helm.Data struct, one could for example overwrite existing keys by duplication and create new keys.

It's just that the code would look much simpler without the recursion. From a threat model point of view, operators can edit cluster resources, but users can create bundles (for those clusters). So I don't see too much of a problem with permissions? Actually, in such a scenario, where users can't change cluster resources,  users would have to ask the operator to include their cluster specific templates.

I'm more concerned with general failure handling, it's easy to get complex templates wrong the first time, but I guess that would just lead to no deployment.